sound/soc/sof/imx/imx8.c

```diff
@@ -372,7 +372,10 @@ static void imx8_ipc_msg_data(struct snd_sof_dev *sdev,
 			      struct snd_sof_pcm_stream *sps,
 			      void *p, size_t sz)
 {
-	sof_mailbox_read(sdev, sdev->dsp_box.offset, p, sz);
+	if (!sps || !sdev->stream_box.size)
+		sof_mailbox_read(sdev, sdev->dsp_box.offset, p, sz);
+	else
+		sof_mailbox_read(sdev, sdev->stream_box.offset, p, sz);
```

Comment:
What does this have to do with the compressed interface support? 
Does this belong in this PR, or is this an unrelated fix?

Reply:
Actually, this is unrelated. I think it is better to have it in a separate PR.

Reply:
@plbossart, I think I will keep this patch here as it the others wont apply if the order is changed.